From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Sat, 9 Aug 2025 14:52:44 +0900
Subject: [PATCH] thread unsafe chunk map patch


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 5d9dd6feb9a33351de6b9b3d7bceb3e904585e4d..6b6edcb58c8486e15c714d04435e3be4dff14e96 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1769,6 +1769,10 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
                     serverLevelTickingSemaphore.acquire();
                     tasks.add(
                         serverLevel.tickExecutor.submit(() -> {
+                            // Leaf start - thread unsafe chunk map
+                            serverLevel.getChunkSource().fullChunksNonSync.setThread();
+                            serverLevel.moonrise$getChunkTaskScheduler().chunkHolderManager.chunkHoldersNoSync.setThread();
+                            // Leaf end - thread unsafe chunk map
                             ca.spottedleaf.moonrise.common.util.TickThread.ServerLevelTickThread currentThread = (ca.spottedleaf.moonrise.common.util.TickThread.ServerLevelTickThread) Thread.currentThread();
                             currentThread.currentlyTickingServerLevel = serverLevel;
 
@@ -1787,7 +1791,11 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             }
 
             while (!tasks.isEmpty()) {
-                tasks.pop().get();
+                // Leaf start - thread unsafe chunk map
+                ServerLevel serverLevel = tasks.pop().get();
+                serverLevel.getChunkSource().fullChunksNonSync.setThread();
+                serverLevel.moonrise$getChunkTaskScheduler().chunkHolderManager.chunkHoldersNoSync.setThread();
+                // Leaf end - thread unsafe chunk map
             }
         } catch (java.lang.InterruptedException | java.util.concurrent.ExecutionException e) {
             throw new RuntimeException(e); // Propagate exception
