From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Altiami <yoshimo.kristin@gmail.com>
Date: Wed, 5 Mar 2025 13:13:24 -0800
Subject: [PATCH] SparklyPaper: Parallel world ticking

Original project: https://github.com/SparklyPower/SparklyPaper

Commit: 589225495e566c60feb907a3571c1ccba855b6ed

diff --git a/src/main/java/org/dreeam/leaf/command/LeafCommand.java b/src/main/java/org/dreeam/leaf/command/LeafCommand.java
index 8b6f62673a99be8decdadd2e44bd993b603ea11e..09ec7aeb3ea655c08c46725adf9195bda12c50fa 100644
--- a/src/main/java/org/dreeam/leaf/command/LeafCommand.java
+++ b/src/main/java/org/dreeam/leaf/command/LeafCommand.java
@@ -5,7 +5,7 @@ import it.unimi.dsi.fastutil.Pair;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.format.NamedTextColor;
 import net.minecraft.Util;
-//import org.dreeam.leaf.command.subcommands.MSPTCommand;
+import org.dreeam.leaf.command.subcommands.MSPTCommand;
 import org.dreeam.leaf.command.subcommands.ReloadCommand;
 import org.dreeam.leaf.command.subcommands.VersionCommand;
 import org.jetbrains.annotations.NotNull;
@@ -38,13 +38,13 @@ public final class LeafCommand extends Command {
     // subcommand label -> subcommand
     private static final LeafSubcommand RELOAD_SUBCOMMAND = new ReloadCommand();
     private static final LeafSubcommand VERSION_SUBCOMMAND = new VersionCommand();
-    //private static final LeafSubcommand MSPT_SUBCOMMAND = new MSPTCommand(); // TODO: Waiting PWT
+    private static final LeafSubcommand MSPT_SUBCOMMAND = new MSPTCommand();
     private static final Map<String, LeafSubcommand> SUBCOMMANDS = Util.make(() -> {
         final Map<Set<String>, LeafSubcommand> commands = new HashMap<>();
 
         commands.put(Set.of(ReloadCommand.LITERAL_ARGUMENT), RELOAD_SUBCOMMAND);
         commands.put(Set.of(VersionCommand.LITERAL_ARGUMENT), VERSION_SUBCOMMAND);
-        //commands.put(Set.of(MSPTCommand.LITERAL_ARGUMENT), MSPT_SUBCOMMAND);
+        commands.put(Set.of(MSPTCommand.LITERAL_ARGUMENT), MSPT_SUBCOMMAND);
 
         return commands.entrySet().stream()
             .flatMap(entry -> entry.getKey().stream().map(s -> Map.entry(s, entry.getValue())))
